comparators.go

```diff
@@ -0,0 +1,22 @@
+package ipx
+
+import (
+	"errors"
+	"net"
+)
+
+// CmpIP compares two IPs
+func CmpIP(a, b net.IP) int {
+	four := a.To4() != nil
+	if four != (b.To4() != nil) {
+		panic(errors.New("IP versions must be the same"))
+	}
+
+	aInt := to128(a.To16())
+	return aInt.Cmp(to128(b.To16()))
+}
+
+// CmpNet compares two networks, using only the IP, disregarding the mask
+func CmpNet(a, b *net.IPNet) int {
```

Comment:
Still not sure whether these should be pointer or the objects. `ParseCIDR` returns a pointer, so if going directly from there, then you can avoid an explicit dereference.

Reply:
I went back and forth on this, but I'm pretty sure it's supposed to be a pointer.

The type's String method has a pointer receiver, which means fmt.Sprint only functions properly if it's a pointer.